src/Documentation/grains/actor_identity.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: page
+title: Actor Identity
+---
+
+# Actor Identity
+
+In object-oriented environments, the identity of an object is hard to distinguish from a reference to it.
+Thus, when an object is created using new, the reference you get back represents all aspects of its identity except those that map the object to some external entity that it represents.
+
+In distributed systems, object references cannot represent instance identity, since references are typically limited to a single address space.
+That is certainly the case for .NET references.
+Furthermore, a virtual actor must have an identity regardless of whether it is active, so that we can activate it on demand.
+Therefore grains have a primary key.
+The primary key can be either a GUID (A Globally Unique Identifier), a long integer, or a string.
+
+The primary key is scoped to the grain type.
+Therefore, the complete identity of a grain is formed from the actor type and its key.
+
+The caller of the grain decides a long, a GUID, or a string scheme should be used.
+In fact the underlying data is the same, so the schemes can be used interchangeably.
+When a long is used, a GUID is actually created, and padded with zeros.
+
+Situations that require a singleton grain instance, such as a dictionary or registry, benefit from using `0` (a valid GUID) as its key.
+This is merely a convention, but by adhering, it becomes clear at the call site that it is what is going on, as we saw in the first tutorial:
+
+## Using GUIDs
+
+GUIDs are useful when there are several processes that could request a grain, such as a number of web servers in a web farm.
+You don't need to coordinate the allocation of keys, which could introduce a single point of failure in the system, or a system-side lock on a resource which could present a bottleneck.
+There is a very low chance of GUIDs colliding, so they would probably be the default choice when architecting an Orleans system.
+
+Referencing a grain by GUID in client code:
+
+``` csharp
+var grain = GrainClient.GrainFactory.GetGrain<IExample>(Guid.NewGuid());
+```
+
+Retrieving the primary key from grain code:
+
+``` csharp
+public override Task OnActivateAsync()
+{
+    Guid primaryKey = this.GetPrimaryKey();
+    return base.OnActivateAsync();
+}
+```
+
+## Using Longs
+
+A long integer is also available, which would make sense if the grain is persisted to a relational database, where numerical indexes are preferred over GUIDs.
+
+Referencing a grain by long integer in client code:
+
+``` csharp
+var grain = GrainClient.GrainFactory.GetGrain<IExample>(1);
```

Comment:
We should change `GrainClient.GrainFactory` to just `grainFactory` here and in other example code.

For history, there used to be a static class called `GrainClient` which users would use to interact with Orleans, but that static class was removed and replaced with the current approach (build the client using ClientBuilder, etc)

Reply:
Thank you for the change suggestion and the history. It's neat to know how Orleans has evolved. I found and replaced 3 instances in this document and will be looking through the rest of the docs later.